joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/NodeBuilder.scala

```diff
@@ -223,28 +223,31 @@ class NodeBuilder(diffGraph: DiffGraphBuilder) {
     addNodeToDiff(fieldIdentifierNode)
   }
 
-  def numberLiteralNode(number: Int, lineAndColumn: LineAndColumn): nodes.NewLiteral = {
-    numberLiteralNode(number.toString, lineAndColumn)
-  }
-
-  def numberLiteralNode(number: String, lineAndColumn: LineAndColumn): nodes.NewLiteral = {
+  def literalNode(string: String, dynamicTypeHint: Option[String], lineAndColumn: LineAndColumn): nodes.NewLiteral = {
     val literalNode = nodes
       .NewLiteral()
-      .code(number)
+      .code(string)
       .typeFullName(Constants.ANY)
+      .dynamicTypeHintFullName(dynamicTypeHint.toList)
       .lineNumber(lineAndColumn.line)
       .columnNumber(lineAndColumn.column)
     addNodeToDiff(literalNode)
   }
 
   def stringLiteralNode(string: String, lineAndColumn: LineAndColumn): nodes.NewLiteral = {
-    val literalNode = nodes
-      .NewLiteral()
-      .code(string)
-      .typeFullName(Constants.ANY)
-      .lineNumber(lineAndColumn.line)
-      .columnNumber(lineAndColumn.column)
-    addNodeToDiff(literalNode)
+    literalNode(string, Some(Constants.builtinStrType), lineAndColumn)
```

Comment:
I think we also need to consider `bytes` literals somewhere around here. (Strings and bytes APIs have some overlap but are different.)